include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,819 @@
+ * with status code except ODP_COMP_ERR_OUT_OF_SPACE.
+ *
+ * for compression + hash, call returns with hash appended to the end of
+ * last processed chunk of data.
+ * User should compute processed data len = total output len - digest_len, where
+ * digest_len queried through odp_comp_hash_alg_capability().
+ *
+ * @param param[in]         Operation parameters.
+ * @param result[out]       Result of operation.
+ *
+ * @retval 0 on success
+ * @retval <0 on failure
+ */
+int odp_comp_compress(odp_comp_op_param_t   *param,
+		      odp_comp_op_result_t  *result);
+
+/**
+ * Comp compress data in asynchronous mode.
+ *
+ * If session is created in ODP_COMP_ASYNC mode, event will be queued
+ * to completion queue. Application should monitor ODP_EVENT_PACKET with
+ * subtype ODP_EVENT_PACKET_COMP on queue.
+ *
+ * If session is created in ODP_COMP_SYNC mode, call fails with status
+ * code ODP_COMP_ERR_NOT_SUPPORTED.
+ *
+ * For compression + hash, call returns with hash appended to the end of
+ * last processed chunk of data.
+ * User should compute processed data len = total output len - digest_len, where
+ * digest_len queried through odp_comp_hash_alg_capability().
+ *
+ * If operation updates result structure with status
+ * ODP_COMP_ERR_OUT_OF_SPACE then application
+ * should call API again with valid output buffer (and no-more input)
+ * until call completes with any other error code.
+ * Please note it is always recommended that application using async mode,
+ * provide sufficiently large buffer size to avoid ODP_COMP_ERR_OUT_OF_SPACE.
+ * Else it is recommended that application maintain relevant context
+ * with respect to each input processing request to correctly identify
+ * its corresponding enqueued event.
+ *
+ * @param param[in]          Operation parameters.
+ *
+ * @retval 0 on success
+ * @retval <0 on failure
+ */
+int odp_comp_compress_enq(odp_comp_op_param_t *param);
+
+  /**
+   * Comp decompress data in synchronous mode
+   *
+   * If session is created in ODP_COMP_SYNC mode, this call wait for operation
+   * to complete and update result at output
+   *
+   * If session is created in ODP_COMP_ASYNC mode, this call fails and update
+   * status code ODP_COMP_ERR_NOT_SUPPORTED.
+   *
+   * If operation returns ODP_COMP_ERR_OUT_OF_SPACE, then application should
+   * call API again with valid output buffer (and no-more input) until call
+   * completes with status code except ODP_COMP_ERR_OUT_OF_SPACE.
+   *
+   * for decompression + hash, call returns with hash appended to the end of
+   * last processed chunk of data.User should compute processed data len =
+   * total output len - digest_len, where digest_len queried through
+   * odp_comp_hash_alg_capability().
+   *
+   * @param param[in]          Operation parameters.
+   * @param result[out]        Result of operation.
+   *
+   * @retval 0 on success
+   * @retval <0 on failure
+   */
+int odp_comp_decomp(odp_comp_op_param_t   *param,
```

Comment:
For `compress`/`decomp` that would be a matter of taste.
When I was asking about two functions I was talking about separate compression/decompression funcions, not about sync/async case.